jenkins-agent/base/ansible/playbooks/tasks/common.yaml

```diff
@@ -9,94 +13,100 @@
     mv kustomize /usr/local/bin/
   when: required is defined and required | bool
   retries: 3
   delay: 5
 
 - name: Install Java 11
   shell: |
-    curl -fLo jdk-{{ jdk_11_version }}.tar.gz https://aka.ms/download-jdk/microsoft-jdk-{{ jdk_11_version }}-linux-{{ arch_x64 or arch_aarch64 }}.tar.gz &&
-    tar zxf jdk-{{ jdk_11_version }}.tar.gz &&
-    mv jdk-{{ jdk_11_version_dir }}+9 /opt/java-{{ jdk_11_version_dir }} &&
+    curl -fLo jdk-{{ jdk_11_version }}.tar.gz https://aka.ms/download-jdk/microsoft-jdk-{{ jdk_11_version }}-linux-{{ arch_x64 or arch_aarch64 }}.tar.gz &&  
+    tar zxf jdk-{{ jdk_11_version }}.tar.gz &&  
+    mv jdk-{{ jdk_11_version_dir }}+9 /opt/java-{{ jdk_11_version_dir }} &&  
     rm jdk-{{ jdk_11_version }}.tar.gz
   when: required is defined and required | bool
   retries: 3
   delay: 5
 - name: Install sonar scanner
   shell: |
-    curl -Lo /tmp/sonar_scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-{{ sonar_scanner_version }}.zip &&
-    unzip /tmp/sonar_scanner.zip -d /opt/ &&
-    mv /opt/sonar-scanner-{{ sonar_scanner_version }} /opt/sonar-scanner &&
-    sed -i '1 a JAVA_HOME=/opt/java-{{ jdk_11_version_dir }}' /opt/sonar-scanner/bin/sonar-scanner &&
-    echo 'export PATH=/opt/sonar-scanner/bin:$PATH' >> /etc/profile.d/sonar-scanner.sh &&
+    curl -Lo /tmp/sonar_scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-{{ sonar_scanner_version }}.zip &&  
+    unzip /tmp/sonar_scanner.zip -d /opt/ &&  
+    mv /opt/sonar-scanner-{{ sonar_scanner_version }} /opt/sonar-scanner &&  
+    sed -i '1 a JAVA_HOME=/opt/java-{{ jdk_11_version_dir }}' /opt/sonar-scanner/bin/sonar-scanner &&  
+    echo 'export PATH=/opt/sonar-scanner/bin:$PATH' >> /etc/profile.d/sonar-scanner.sh &&  
     sonar-scanner --version
   when: required is defined and required | bool
   retries: 3
   delay: 5
 
 - name: Install Go
   shell: |
-    wget https://golang.google.cn/dl/go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz
-    tar -xvf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz
-    rm -rf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz
-    mv go /usr/local/go
+    wget https://golang.google.cn/dl/go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz  
+    tar -xvf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz  
+    rm -rf go{{ go_version }}.linux-{{ arch_amd64 or arch_arm64 }}.tar.gz  
+    mv go /usr/local/go  
     go version
   when: go_version is defined and go_version != ""
   retries: 3
   delay: 5
 
 - name: Install Node.js
   shell: |
-    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.0/install.sh | bash
-    . "$NVM_DIR/nvm.sh"
-    nvm install {{ node_version }}
-    nvm alias default {{ node_version }}
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.0/install.sh | bash  
```

Comment:
imo nvm is not the best practice to install nodejs in container. 

It is more like `virtualenv` in python, a development environment management tool to allow multiple versions coexist in a host. But in container you have no such requirement, you just pull another image ...